include/exadg/aero_acoustic/driver.cpp

```diff
@@ -135,26 +123,50 @@ template<int dim, typename Number>
 void
 Driver<dim, Number>::solve()
 {
-  set_start_time();
+  std::pair<bool, dealii::Timer> timer = std::make_pair(false, dealii::Timer());
 
-  synchronize_time_step_size();
+  set_start_time();
 
   AssertThrow(std::abs(application->fluid->get_parameters().end_time -
                        application->acoustic->get_parameters().end_time) < 1.0e-12,
               dealii::ExcMessage("Acoustic and fluid simulation need the same end time."));
 
-  while(not acoustic->time_integrator->finished())
+  while(not fluid->time_integrator->finished())
   {
+    if(timer.first == false and acoustic->time_integrator->started())
+    {
+      timer.first = true;
+      timer.second.restart();
+    }
+
+    // To check if acoustics starts during the following sub-stepping sweep we
+    // can not simply check acoustic->time_integrator->started(). Instead we
+    // compute this information as follows:
+    const bool acoustic_starts_during_present_timestep =
+      fluid->time_integrator->get_next_time() + fluid->time_integrator->get_time_step_size() >
+      application->acoustic->get_parameters().start_time;
+
     // The acoustic simulation uses explicit time-stepping while the fluid solver
     // uses implicit time-stepping. Therefore, we advance the acoustic solver to
     // t^(n+1) first and directly use the result in the fluid solver.
-    if(acoustic->time_integrator->started())
+    if(acoustic_starts_during_sub_stepping)
```

Comment:
This name probably needs to be adjusted?

Reply:
[fix compilation](https://github.com/exadg/exadg/pull/645/commits/8be5473532ecc37cd1fcb15a56207898a8759850)